lib/system/dyncalls.nim

```diff
@@ -149,6 +149,20 @@ elif defined(windows) or defined(dos):
       if result != nil: return
     procAddrError(name)
 
+  proc nimLoadLibraryError(path: string) =
+    # carefully written to avoid memory allocation:
+    when defined(guiapp):
```

Comment:
when defined(windows) and defined(guiapp)  

and then you can merge the two nimLoadLibraryError implementations.